src/debug_clang.h

```diff
@@ -0,0 +1,85 @@
+#ifndef JUCI_DEBUG_CLANG_H_
+#define JUCI_DEBUG_CLANG_H_
+
+#include <boost/filesystem.hpp>
+#include <unordered_map>
+#include <lldb/API/SBDebugger.h>
+#include <lldb/API/SBListener.h>
+#include <lldb/API/SBProcess.h>
+#include <thread>
+#include <mutex>
+
+namespace Debug {
+  class Clang {
```

Comment:
I saw some more abstraction when I got further down. But I'm still a bit confused. If I were to add a debugger for a different language, or syntax highlighting for that matter, how would I approach?